src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpHeaders.cs

```diff
@@ -235,14 +241,19 @@ public override string ToString()
 
             var vsb = new ValueStringBuilder(stackalloc char[512]);
 
-            if (_headerStore is Dictionary<HeaderDescriptor, object> headerStore)
+            if (GetEntries() is HeaderEntry[] entries)
```

Comment:
Why `is HeaderEntry[] entries` and not simple null check?

Reply:
What do you mean by "simple null check"? `GetEntries() is { } entries)`?
If so, readability at the callsite

Reply:
Yep, on the second look this looks fine